x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -123,6 +125,32 @@ const nonAggregatableDatasetsRoute = createDatasetQualityServerRoute({
   },
 });
 
+const degradedFieldsRoute = createDatasetQualityServerRoute({
+  endpoint: 'GET /internal/dataset_quality/data_streams/{dataStream}/degraded_fields',
+  params: t.type({
```

Comment:
**nit**: this type might be better located in [x-pack/plugins/observability_solution/dataset_quality/common/api_types.ts](https://github.com/elastic/kibana/pull/183934/files#diff-d6e6732e8f65c1856abbb34628a002d63665261c513b1bbe58291c25f7e74ccf) along with the response type.

Reply:
In APM we found it really helpful to keep the types right with the endpoints as you can clearly see how the endpoint looks like in one place. But there we had our own Custom HTTP Client which would mean a lot of TS level validations would be done by it, which we don't have here. Hence we have types spread across multiple locations at the moment. Like below

1. With server routes as commented here
2. Inside [x-pack/plugins/observability_solution/dataset_quality/common/api_types.ts](https://github.com/elastic/kibana/pull/183934/files#diff-d6e6732e8f65c1856abbb34628a002d63665261c513b1bbe58291c25f7e74ccf)
3. Inside [x-pack/plugins/observability_solution/dataset_quality/common/data_streams_stats/types.ts](https://github.com/elastic/kibana/blob/919b207a3d5011f058f26533f2120971a25ccded/x-pack/plugins/observability_solution/dataset_quality/common/data_streams_stats/types.ts)

We will have to plan a tech debt to refactor them into more reusable and reasonable blocks.